.github/workflows/codesee-arch-diagram.yml

```diff
@@ -0,0 +1,84 @@
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '25 0/12 * * 1-5' # Build once every 12 hours, starting at 25 minutes past the hour (UTC)
```

Comment:
Should we do this so often for this repo?

I am a little confused what this event is targeting... We want to build the map for each PR, but only if:
a) it has been open long enough to get past the 12h25min
b) Rebuild every 12h?

Reply:
> Should we do this so often for this repo?

This repo has the most impact on our end-users, and while Kris has done incredibly well in keeping the code very performant, there is scope to improve per the Codefactor rating.

I have a few things in the works that should help, but I would like a quick overview of the code structure due to a lack of time on my part. This action should help that. 

I may remove it when I am done with it.

> I am a little confused what this event is targeting... 

I would not blame you.

So the events & conditions are:

1. `pull_request_target`

   "Build the CodeSee map" for every change in the PR (usually very fast because it's essentially a diff). 

   This happens for every PR, for every push on the PR, without any waiting.

2. On a `schedule` 

   "Build the CodeSee map" for the default branch every 12 hours (25 mins past the hour).

3. On a `workflow_dispatch`

   Let someone with access run an ad-hoc build anytime they want. This is called dispatching a workflow event in GitHub terminology.

4. Do not build for bots.